synapse/handlers/federation.py

```diff
@@ -1004,16 +1013,47 @@ async def maybe_backfill(self, room_id, current_depth):
         sorted_extremeties_tuple = sorted(extremities.items(), key=lambda e: -int(e[1]))
         max_depth = sorted_extremeties_tuple[0][1]
 
-        # We don't want to specify too many extremities as it causes the backfill
-        # request URI to be too long.
-        extremities = dict(sorted_extremeties_tuple[:5])
-
-        if current_depth > max_depth:
+        # If we're approaching an extremity we trigger a backfill, otherwise we
+        # no-op.
+        if current_depth - 2 * limit > max_depth:
```

Comment:
This looks reasonable, but I'm struggling with the 2x here.

Reply:
Will add a comment